# Plot: 
1. [circlize](https://github.com/jokergoo/circlize): circular visualization in R: 
2. [PyGWalker](https://kanaries.net/pygwalker): turns your data into interactive visualization apps with one line of code, and enables you to share your apps with one click.
# New models:
1. [RUDOLFV](https://arxiv.org/pdf/2401.04079): A foundation model by pathologists for pathologists
2. [HistoSSLscaling](https://github.com/owkin/HistoSSLscaling): Scaling Self-Supervised Learning for Histopathology with Masked Image Modeling
3. [DEPLOY](https://zenodo.org/records/10888478)): An integrated deep learning model for predicting DNA methylation and tumor types from H&E images
4. [CODA](https://github.com/ashleylk/CODA): 3D genomic mapping reveals multifocality of human pancreatic precancers
5. [Pythia-160m model and Cell2Sentence](https://huggingface.co/vandijklab/pythia-160m-c2s)
6. [review](https://www.apoorva-srinivasan.com/plms/): a review on protein language models
7. [His2Cell](https://github.com/Weiqin-Zhao/Hist2Cell): a Vision Graph-Transformer framework that accurately predicts fine-grained cell type abundances directly from histology images.
# Constrastive learning:
1. [TransPath](https://github.com/Xiyue-Wang/TransPath): Transformer-based unsupervised contrastive learning for histopathological image classification
1. [T-cells](https://www.science.org/doi/10.1126/sciadv.adk4670): Self-supervised learning of T cell receptor sequences exposes core properties for T cell membership
2. [scDiffusion](https://github.com/EperLuo/scDiffusion): conditional generation of high-quality single-cell data using diffusion model
3. [Proust](https://www.biorxiv.org/content/10.1101/2024.02.02.578662v1): graph-based using spatially resolved transcriptomics and immunofluorescence images https://github.com/JianingYao/proust
4. [Spacial-ConGR](https://www.sciencedirect.com/science/article/pii/S200103702400120X):  A contrastive learning approach to integrate spatial transcriptomics and histological images https://github.com/YuLin-code/Spatial-ConGR
5. [SCLSC](https://github.com/yaozhong/SCLSC): Predicting cell types with supervised contrastive learning on cells and their types
6. [MSI](https://github.com/LabLaskin/MSI-self-supervised-clustering): Contrastive learning model to find co-localisation of m/z peaks in MALDI data
