# Plot and tools:
1. [circlize](https://github.com/jokergoo/circlize): circular visualization in R: 
2. [PyGWalker](https://kanaries.net/pygwalker): turns your data into interactive visualization apps with one line of code, and enables you to share your apps with one click.
3. [BioCypher](https://biocypher.org/): BioCypher is the simplest way to create an AI-enabled knowledge graph for biomedical (or other) tasks.
# New models:
1. [RUDOLFV](https://arxiv.org/pdf/2401.04079): A foundation model by pathologists for pathologists
2. [HistoSSLscaling](https://github.com/owkin/HistoSSLscaling): Scaling Self-Supervised Learning for Histopathology with Masked Image Modeling
3. [DEPLOY](https://zenodo.org/records/10888478): An integrated deep learning model for predicting DNA methylation and tumor types from H&E images
4. [CODA](https://github.com/ashleylk/CODA): 3D genomic mapping reveals multifocality of human pancreatic precancers
5. [Pythia-160m model and Cell2Sentence](https://huggingface.co/vandijklab/pythia-160m-c2s)
6. [review](https://www.apoorva-srinivasan.com/plms/): a review on protein language models
7. [His2Cell](https://github.com/Weiqin-Zhao/Hist2Cell): a Vision Graph-Transformer framework that accurately predicts fine-grained cell type abundances directly from histology images. https://www.biorxiv.org/content/10.1101/2024.02.17.580852v1.full.pdf
8. [BioChatter](https://biochatter.org/): A framework for deploying, testing, and evaluating conversational AI models in the biomedical domain.
9. [](https://github.com/OxWearables/ssl-wearables): Multi-task self-supervised learning for wearables
1. [UniFMIR](https://www.nature.com/articles/s41592-024-02244-3): Pretraining a foundation model for generalizable fluorescence microscopy-based image restoration
1. [UNI](https://www.nature.com/articles/s41591-024-02857-3): Towards a general-purpose foundation model for computational pathology https://github.com/mahmoodlab/UNI
2. [TORCH](https://www.nature.com/articles/s41591-024-02915-w): Prediction of tumor origin in cancers of unknown primary origin with cytology-based deep learning

$$
x = \frac{{-b \pm \sqrt{{b^2 - 4ac}}}}{{2a}}
$$

# Constrastive learning:
1. [TransPath](https://github.com/Xiyue-Wang/TransPath): Transformer-based unsupervised contrastive learning for histopathological image classification
1. [T-cells](https://www.science.org/doi/10.1126/sciadv.adk4670): Self-supervised learning of T cell receptor sequences exposes core properties for T cell membership
2. [scDiffusion](https://github.com/EperLuo/scDiffusion): conditional generation of high-quality single-cell data using diffusion model
3. [Proust](https://www.biorxiv.org/content/10.1101/2024.02.02.578662v1): graph-based using spatially resolved transcriptomics and immunofluorescence images https://github.com/JianingYao/proust
4. [Spacial-ConGR](https://www.sciencedirect.com/science/article/pii/S200103702400120X):  A contrastive learning approach to integrate spatial transcriptomics and histological images https://github.com/YuLin-code/Spatial-ConGR
5. [SCLSC](https://github.com/yaozhong/SCLSC): Predicting cell types with supervised contrastive learning on cells and their types
6. [MSI](https://github.com/LabLaskin/MSI-self-supervised-clustering): Contrastive learning model to find co-localisation of m/z peaks in MALDI data
7. [CONCH](https://www.nature.com/articles/s41591-024-02856-4): A visual-language foundation model for computational pathology
8. [VOLTA](https://www.nature.com/articles/s41467-024-48062-1): an enVironment-aware cOntrastive ceLl represenTation leArning for histopathology. https://github.com/AIMLab-UBC/VOLTA
9. [ResNet*](https://www.cell.com/heliyon/pdf/S2405-8440(24)00125-7.pdf): Enhancing breast cancer classification via histopathological image analysis: Leveraging self-supervised contrastive learning and transfer learning.
10. [DCLGAN](https://www.nature.com/articles/s41598-024-52833-7): Dual contrastive learning based image-to-image translation of unstained skin tissue into virtually stained H&E images
11. [SCLSC](https://www.nature.com/articles/s41598-023-50185-2): Predicting cell types with supervised contrastive learning on cells and their types.
12. [Constrastive-sc](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04210-8): Contrastive self-supervised clustering of scRNA-seq data
1. [SimCLR tutorial notebook](https://uvadlc-notebooks.readthedocs.io/en/latest/tutorial_notebooks/JAX/tutorial17/SimCLR.html)
1. [spatiAlign](https://github.com/STOmics/Spatialign): An Unsupervised Contrastive Learning Model for Data Integration of Spatially Resolved Transcriptomics

# LLMs and Generative AI:
1. [LLM Visualization](https://bbycroft.net/llm)
# Other learning materials:
1. [Tutorial on Diffusion Models for Imaging and Vision](https://arxiv.org/pdf/2403.18103)
2. [Melanoma](https://emedicine.medscape.com/article/1295718-overview?form=fpf)
